src/java.base/share/classes/java/net/Socket.java

```diff
@@ -620,6 +621,8 @@ void setConnected() {
      * @param   endpoint the {@code SocketAddress}
      * @throws  IOException if an error occurs during the connection, the socket
      *          is already connected or the socket is closed
+     * @throws  UnknownHostException if the endpoint is an unresolved
+     *          {@link InetSocketAddress}, the socket is closed
```

Comment:
It might be better to drop ", the socket is closed" from the exception description as it may be mis-read to mean that it gets thrown when the socket is closed.

Reply:
Fixed in 424fc8ccd3a495a49f859f0cfc563cf463b3efeb.